core/control-plane/contract/src/main/java/org/eclipse/dataspaceconnector/contract/offer/ContractOfferServiceImpl.java

```diff
@@ -60,10 +63,22 @@ public Stream<ContractOffer> queryContractOffers(ContractOfferQuery query) {
         var agent = agentService.createFor(query.getClaimToken());
 
         return definitionService.definitionsFor(agent, query.getRange())
+                // ignore definitions with missing criteria (only if not SELECT_ALL)
+                .filter(definition -> (definition.getSelectorExpression().getCriteria() != null &&
```

Comment:
I think this filter does nothing, as when the `criteria` field is empty, then the `selectorExpression` equals to `SELECT_ALL`. Because `SELECT_ALL` means no criterion are provided.
(this passes:)
```java
    @Test
    void equality() {
        assertThat(AssetSelectorExpression.Builder.newInstance().build()).isEqualTo(AssetSelectorExpression.SELECT_ALL);
    }
```

Furthermore, the `selectorExpression` of the definition cannot be null, neither the `criteria` list inside it

Reply:
Still, either I don't get it or I don't agree. 

> I would also remove the SELECT_ALL constant as it is not used by anyone, it could be replaced by a isSelectAll() method on AssetSelectorExpression that checks if the criteria list is empty or not.

We should not follow common DB querying here, as accidentally missing criteria could result in an information leak (what is critically in terms of data sovereignty). So we should not interpret missing criteria as "select all".. what would simplify the implementation from a logically thinking, I agree, but not represent our business logic.

> Because SELECT_ALL means no criterion are provided.

This is also not true because both properties can be set in parallel, don't they? There is no such logic in the `AssetSelectorExpression`, nor is it needed. In my understanding, `SELECT_ALL` has just a higher priority then the criteria list as it must explicitly being set.

Reply:
> Furthermore, the selectorExpression of the definition cannot be null, neither the criteria list inside it

Is this a suggestion or a fact? Because I checked and you're right, a expression cannot be null, still there is currently no logic that checks whether the criteria list is empty or whether one of `SELECT_ALL` or the criteria list is null or empty. Or did I miss anything?